### PR TITLE
Fixing first tap Card behavior

### LINKED ISSUE #1015: Fix flipping card when learning
**Describe the bug**
Card doesn't flip during learning

**To Reproduce**
1. Open learning of cards
1. Click on the card to flip
1. Press 'know' or 'don't know' button
1. Click on the next card to flip
1. See error

**Expected behavior**
Card flips smoothly on click

ezgif com-optimize

### PR DESCRIPTION
**Pull request by jddeep**
Fixing first tap Card behavior.

Fixes #1015

### CODE REVIEW

**Comment by jddeep** [comment]
@ksheremet I was trying to `reset` the controller using some logic. But It is that `widget.newcardcontroller` is always true in `cardDisplayWidget` once one new card comes. And as the former is final I am unable to toggle the value.

**Comment by jddeep** [comment]
@ksheremet Your thoughts?

File: flutter/lib/views/helpers/card_display_widget.dart [hunk 1]
```diff
@@ -12,14 +12,16 @@ class CardDisplayWidget extends StatefulWidget {
   final Color backgroundColor;
   final bool isMarkdown;
   final onFlipCallback onFlip;
+  final bool newCardController;
```
Comment (ksheremet) [inline flutter/lib/views/helpers/card_display_widget.dart]: @jddeep 

Please consider ``` didUpdateWidget``` to check that card is new. No need to create extra param to fix this issue

**Comment by ksheremet** [comment]
@jddeep 
Please do not forget to run the test
/Users/travis/build/dasfoo/delern/flutter/test/card_display_widget_test.dart: Display card

**Comment by jddeep** [comment]
Yes @ksheremet I rand the test. It has passed successfully.

File: flutter/lib/views/helpers/card_display_widget.dart (hunk 1 above)
Comment (jddeep) [reply flutter/lib/views/helpers/card_display_widget.dart]: @ksheremet Made the changes. Were you talking about this? But our purpose is still not resolved.

**Comment by jddeep** [comment]
@ksheremet Issue fixed! Please review.

File: flutter/lib/views/helpers/card_display_widget.dart [hunk 2]
```diff
@@ -42,6 +46,15 @@ class _CardDisplayWidgetState extends State<CardDisplayWidget>
     ]).animate(_controller);
   }
 
+  @override
+  void didUpdateWidget(CardDisplayWidget oldWidget) {
+    if (!(cardText == oldWidget.front)) {
+      _controller.reset();
+      cardText = oldWidget.front;
+    }
+    super.didUpdateWidget(oldWidget);
```
Comment (ksheremet) [inline flutter/lib/views/helpers/card_display_widget.dart]: @jddeep 

Why did you use ```String cardText;```? 
How to call new widget from ```didUpdateWidget``` without ```cardText```?

**Comment by jddeep** [comment]
@ksheremet Fixed. Please review.

File: flutter/lib/views/helpers/card_display_widget.dart [hunk 3]
```diff
@@ -42,6 +42,14 @@ class _CardDisplayWidgetState extends State<CardDisplayWidget>
     ]).animate(_controller);
   }
 
+  @override
+  void didUpdateWidget(CardDisplayWidget oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (!(oldWidget.front == widget.front)) {
```
Comment (ksheremet) [inline flutter/lib/views/helpers/card_display_widget.dart]: @jddeep 

How about ```oldWidget.front != widget.front```? It is better for readability.

**Comment by jddeep** [comment]
@ksheremet Did the changes.

**Pull request merged**